plugin/trino-exasol/src/main/java/io/trino/plugin/exasol/ExasolClient.java

```diff
@@ -0,0 +1,408 @@
+        }
+        if (type == DOUBLE) {
+            return WriteMapping.doubleMapping("double precision", doubleWriteFunction());
+        }
+
+        if (type instanceof DecimalType decimalType) {
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.objectMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+
+        if (type instanceof CharType charType) {
+            if (charType.getLength() > EXASOL_CHAR_MAX_CHARS) {
+                throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName() + ", Exasol supports max length " + EXASOL_VARCHAR_MAX_CHARS + " for CHAR type");
+            }
+            return WriteMapping.sliceMapping("char(" + charType.getLength() + ")", charWriteFunction());
+        }
+
+        if (type instanceof VarcharType varcharType) {
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                dataType = "varchar(" + EXASOL_VARCHAR_MAX_CHARS + ")";
+            }
+            else if (varcharType.getBoundedLength() > EXASOL_VARCHAR_MAX_CHARS) {
+                throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName() + ", Exasol supports max length " + EXASOL_VARCHAR_MAX_CHARS + " for VARCHAR type");
+            }
+            else {
+                dataType = "varchar(" + varcharType.getBoundedLength() + ")";
+            }
+            return WriteMapping.sliceMapping(dataType, varcharWriteFunction());
+        }
+
+        if (type == DATE) {
+            return WriteMapping.longMapping("date", dateWriteFunctionUsingLocalDate());
+        }
+
+        if (type instanceof TimestampType timestampType) {
+            if (timestampType.getPrecision() != EXASOL_TIMESTAMP_PRECISION) {
+                throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName() + ". Exasol supports timestamps only with precision 3.");
+            }
+            return WriteMapping.longMapping("timestamp", timestampWriteFunctionUsingSqlTimestamp(TIMESTAMP_MILLIS));
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Unsupported column type: " + type.getDisplayName());
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginCreateTable(ConnectorSession session, ConnectorTableMetadata tableMetadata)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating tables with data");
+    }
+
+    @Override
+    public JdbcOutputTableHandle beginInsertTable(ConnectorSession session, JdbcTableHandle tableHandle, List<JdbcColumnHandle> columns)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support inserts");
+    }
```

Comment:
Supporting CREATE TABLE statement without INSERT statement doesn't make sense in my opinion.